### PR TITLE
fix(runner): empty runner file queue when killing the bots process

### PR DESCRIPTION
**Pull request by rtsun95**
Empty the files queue when killing the bots. It will otherwise still run following files.

### CODE REVIEW

**Comment by yoannmoinet** [comment]
Even if not armful as it gets reset in the `run` function.
It makes sense to empty it there.

**Comment by rtsun95** [comment]
Update the PR by killing the PhantomJS process before killing CasperJS. Also, stop killing child when a suite ends because CasperJS exits before, so killing it again leads to an error.

File: src/runner.js [hunk 1]
```diff
@@ -182,10 +182,25 @@ Runner.prototype.killChild = function killChild (uuid) {
     }
 
     if (typeof child.kill === 'function') {
-        child.kill('SIGTERM');
+        var isWin = /^win/.test(process.platform);
+        var killCommand = killCommand = 'kill -9 ' + child.PhantomPID;
```
Comment (yoannmoinet) [inline src/runner.js]: why the double assignation `var killCommand = killCommand ...`?
Is there a global variable `killCommand`?

File: src/runner.js [hunk 2]
```diff
@@ -182,10 +182,25 @@ Runner.prototype.killChild = function killChild (uuid) {
...
+            if (error !== null) {
+                console.log(chalk.bgRed(' -[ ERROR ]- '), error);
+                return;
+            }
+            child.kill('SIGTERM');
+        });
     }
 
-    this.cleanChild(uuid);
+    if (child.killed) {
```
Comment (yoannmoinet) [inline src/runner.js:209]: is it part of the child's API?

File: src/runner.js [hunk 3]
```diff
@@ -182,10 +182,25 @@ Runner.prototype.killChild = function killChild (uuid) {
...
     if (typeof child.kill === 'function') {
-        child.kill('SIGTERM');
+        var isWin = /^win/.test(process.platform);
+        var killCommand = killCommand = 'kill -9 ' + child.PhantomPID;
+
+        if (isWin) {
+            killCommand = 'taskkill /PID ' + child.PhantomPID + ' /F';
+        }
+
+        exec(killCommand, function (error, stdout, stderr) {
```
Comment (yoannmoinet) [inline src/runner.js:195]: can you comment on the process and why you have to do all of this please?

File: src/runner.js [hunk 4]
```diff
@@ -347,7 +362,12 @@ Runner.prototype.bindChild = function bindChild (child) {
     child.on('close', this.end.bind(this, child));
 
     child.stdout.on('data', function (data) {
-        console.log(chalk.blue(' -[ child ]- '), data.toString());
+        var output = data.toString();
+        if (output.indexOf('PhantomJS PID: ') > -1) {
```
Comment (yoannmoinet) [inline src/runner.js:378]: can you comment this also please?

**Comment by yoannmoinet** [comment]
Good stuff... it's really too bad we have to hard-kill the PhantomJS process this way.

Is it working correctly for your use-cases? 
Do you actually see your children getting killed (do not read this out of context...)

I think there are just minor updates and precisions, it will be merged then.

Thanks !

File: src/runner.js (hunk 1 above)
Comment (rtsun95) [inline src/runner.js]: uh, the double assignment should not be there...going to fix it

File: src/runner.js (hunk 2 above)
Comment (rtsun95) [inline src/runner.js:209]: It's not on the Node.js docs, but I found this property get set to `true` when the child process is explicitly killed by the `kill()` command, which is what we want to do with this function.

**Comment by rtsun95** [comment]
Changes done!!
It works well for what I need to do. Children indeed get killed and stop running when `damon.kill()` is called.

**Comment by yoannmoinet** [comment]
merged (FINALLY)

**Pull request merged**